### PR TITLE
Implement train search

### PR DESCRIPTION
**Pull request by DRSchlaubi**
Co-authored-by: Emma Böcker <stckoverflw@gmail.com>

### CODE REVIEW

**Comment by NyCodeGHG** [comment]
I'm a bit unsure if it's good to introduce a dependency on hafas if this projects goal was to use vendo as the data source.
Also this does not compile?

File: src/requests/hafas/journey_match.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  id: string
+  destination: Station,
+  origin: Station,
+  line: {
+    name: string
+  }
+}
+
+export default async function journeyMatch(input: string, date: Date): Promise<Array<Trip>> {
+  const response = await fetch(`${getBaseUrl()}/api/hafas/journey_match?q=${input}&date=${date}`, {
```
Comment (NyCodeGHG) [inline src/requests/hafas/journey_match.ts:17]: getBaseUrl() is server side code, which won't work on the client

File: src/pages/api/hafas/journey_match.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+const client = createClient(dbProfile, "RailBoard")
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const {q} = req.query
+  try {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```
Comment (NyCodeGHG) [inline src/pages/api/hafas/journey_match.ts:13]: 🥴

**Comment by NyCodeGHG** [comment]
you don't need to delete `node_modules`, just running `pnpm install` is fine

File: src/requests/hafas/journey_match.ts (hunk 1 above)
Comment (DRSchlaubi) [reply src/requests/hafas/journey_match.ts:17]: It does work and @StckOverflw's code does it as well

**Comment by StckOverflw** [comment]
Typing something into the search bar does not work. It's stuck in the loading state, first thing I saw in the vercel console is this 

image

**Comment by DRSchlaubi** [comment]
base.json ist part of hafas-client

**Comment by StckOverflw** [comment]
The error on vercel is still happening for some reason

**Review by StckOverflw — APPROVED** [review]

**Pull request merged**